pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -918,6 +926,35 @@ func (ctrl *PersistentVolumeController) updateVolumePhaseWithEvent(volume *v1.Pe
 	return newVol, nil
 }
 
+// assignDefaultStorageClass looks up all storage classes and attempts to assign a storage class to the given claim.
+// The storage class assigned is always the default one and if there is none or there are more defaults the claim
+// will not get updated.
```

Comment:
Can you say a little more about the fact that this will actually update the API (not the passed-in object) and then return the net-new object?  I did not expect that from this comment.

Reply:
Done - changed the description to make it more clear.